modules/protocol/inspircd.c

```diff
@@ -208,30 +284,46 @@ channel_metadata_sts(struct channel *c, const char *key, const char *value)
 	sts(":%s METADATA %s %s :%s", ME, c->name, key, value);
 }
 
-static bool
-check_jointhrottle(const char *value, struct channel *c, struct mychan *mc, struct user *u, struct myuser *mu)
+/* Checks for an invalid integer parameter
+ * Returns -1 on overflow or error
+ */
+static int
+param_int(const char *value, int len)
 {
-	const char *p, *arg2;
+	const char *p = value;
+	long long ret;
 
-	p = value;
-	arg2 = NULL;
+	if(len < 0 || len > 10)
+		return -1;
 
-	while (*p != '\0')
+	for( ; *p != '\0' || (len && (value - p) > len); p++)
```

Comment:
Seems like a good idea to move `const char *p = value` into the first part of the for loop head here – this both avoids an empty for loop initial step and moves the declaration to where it's needed